app/src/main/java/com/android/voyageur/model/assistant/ActivityFromAssistant.kt

```diff
@@ -0,0 +1,94 @@
+package com.android.voyageur.model.assistant
+
+import android.util.Log
+import com.android.voyageur.model.activity.Activity
+import com.android.voyageur.model.activity.ActivityType
+import com.google.firebase.Timestamp
+import com.google.gson.Gson
+import com.google.gson.reflect.TypeToken
+import java.time.ZoneId
+
+/**
+ * We need a separate class for the JSON that the assistant generates since the assistant can
+ * confuse the TimeStamp format (it's less error prone to generate dates and times in a readable .
+ * This class is used to store the data in a format that the assistant can understand.
+ */
+data class ActivityFromAssistant(
+    val title: String = "",
+    val description: String = "",
+    val year: Int = 0,
+    val month: Int = 0,
+    val day: Int = 0,
+    val startTimeHour: Int = 0,
+    val startTimeMinute: Int = 0,
+    val endTimeHour: Int = 0,
+    val endTimeMinute: Int = 0,
+    val estimatedPrice: Double = 0.0,
+    val activityType: String = "" // it's a string because the generativeModel can't generate enums
+)
+
+/**
+ * Extracts activities that the assistant generated from a JSON string.
+ *
+ * @param jsonString the JSON string
+ * @return the list of activities
+ */
+fun extractActivitiesFromAssistantFromJson(jsonString: String): MutableList<ActivityFromAssistant> {
+  val gson = Gson()
+  val activityListType = object : TypeToken<List<ActivityFromAssistant>>() {}.type
+  return gson.fromJson(jsonString, activityListType)
+}
```

Comment:
It could be beneficial to add error handling for cases where the JSON might be malformed or when parsing exceptions occur, like wrapping the JSON parsing logic in a try-catch block.